src/Subscriber/AbstractEnderecoSubscriber.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Endereco\Shopware6Client\Subscriber;
+
+use Endereco\Shopware6Client\Entity\EnderecoAddressExtension\EnderecoAddressExtensionEntity;
+use Endereco\Shopware6Client\Service\EnderecoService;
+use Shopware\Core\Checkout\Customer\Aggregate\CustomerAddress\CustomerAddressEntity;
+use Shopware\Core\Framework\Api\Context\SalesChannelApiSource;
+use Shopware\Core\Framework\Context;
+use Shopware\Core\Framework\DataAbstractionLayer\EntityRepository;
+use Shopware\Core\System\SystemConfig\SystemConfigService;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+abstract class AbstractEnderecoSubscriber implements EventSubscriberInterface
+{
+    protected SystemConfigService $systemConfigService;
+    protected EnderecoService $enderecoService;
+    protected EntityRepository $customerAddressRepository;
+
+    public function __construct(
+        SystemConfigService $systemConfigService,
+        EnderecoService     $enderecoService,
+        EntityRepository    $customerAddressRepository
+    )
+    {
+        $this->enderecoService = $enderecoService;
+        $this->systemConfigService = $systemConfigService;
+        $this->customerAddressRepository = $customerAddressRepository;
+    }
+
+    abstract public static function getSubscribedEvents(): array;
+
+    /**
+     * Checking if current customer address is split.
+     * If not or currently saved Shopware address is different - split it automatically
+     */
+    protected function ensureAddressIsSplit(Context $context, ?CustomerAddressEntity $address = null): void
+    {
+        if (is_null($address)) {
+            return;
+        }
+        /* @var $enderecoAddress EnderecoAddressExtensionEntity */
+        $enderecoAddress = $address->getExtension('enderecoAddress');
+
+        if (!$enderecoAddress && !is_null($address->getCountry())) {
```

Comment:
Right, I have covered case when country is null inside of `CustomerAddressEntity`, now it will automatically fetch country by id. Whole logic is refactored to make it more readable 
![image](https://user-images.githubusercontent.com/22631551/215733908-f8404ae9-eb03-4f63-9214-d426dc160f00.png)